AK/Math.h

```diff
@@ -251,7 +251,8 @@ constexpr void sincos(T angle, T& sin_val, T& cos_val)
         : "=t"(cos_val), "=u"(sin_val)
         : "0"(angle));
 #else
-    __builtin_sincosf(angle, sin_val, cos_val);
+    sin_val = __builtin_sin(angle);
+    cos_val = __builtin_cos(angle);
```

Comment:
Wouldn't regular `sin` (resolves to `AK::sin`) work here? It selects the float/double/long double version correctly, so you won't end up needlessly increasing precision or truncation to double.

You seem to have misunderstood the problem with `__builtin_sincosf`. The issue isn't that it is not available, but instead that it takes pointers as arguments, not references. Furthermore, float/double pointers can't be converted to each other, so you'd have to dispatch the  `sincosf`/`sincos`/`sincosl` builtins based on the argument type like how `CONSTEXPR_STATE` does it.

Reply:
Yes regular sin works. I had that initially and now changed back to that. I don't really understand how to tell what type of argument is expected and what type it should return, so I hope this is correct.